src/builder.rs

```diff
@@ -189,6 +192,38 @@ impl SessionBuilder {
         self
     }
 
+    /// Specify the path to the ssh-agent.
+    ///
+    /// The path provided may use tilde notation (`~`) to refer to the user's
+    /// home directory.
+    ///
+    /// The default is `None`.
+    pub fn ssh_auth_sock(&mut self, ssh_auth_sock: impl AsRef<Path>) -> &mut Self {
+        self.ssh_auth_sock = Some(ssh_auth_sock.as_ref().to_owned().into_boxed_path());
+        self
+    }
+
+    /// Inherit the `SSH_AUTH_SOCK` environment variable from parent.
+    ///
+    /// This is equivalent to:
+    ///
+    /// ```
+    /// use openssh::SessionBuilder;
+    /// use std::{env, path::PathBuf};
+    ///
+    /// let mut builder = SessionBuilder::default();
+    ///
+    /// if let Some(ssh_auth_sock) = env::var_os("SSH_AUTH_SOCK") {
+    ///     builder.ssh_auth_sock(PathBuf::from(ssh_auth_sock));
+    /// }
+    /// ```
+    pub fn inherit_ssh_auth_sock(&mut self) -> &mut Self {
```

Comment:
I'm confused by this. We never call `env_clear` (or is it `clear_env`?) on the `CommandBuilder`, so environment variables should apply to the sub-process when we execute it regardless. Why have a dedicated method for this behavior when it's already what'll be happening?

Reply:
Oops I forgot this when I saw the comment https://github.com/openssh-rust/openssh/issues/108#issue-1504346738 

I suppose I should remove this method then, since @TaaviE apparently has encountered other issues.

Reply:
Done.